File: Facit/ClientApp/src/models/transaction.ts
import { Project } from "./project";
import { Person } from "./person";
import { Currency } from "./currency";
import { TransactionItem } from './transactionItem';

export interface Transaction {
    id: number;
    project: Project;
    description: string;
    when: Date;
    createdBy: Person;
    currencyLocal: Currency;
    participants: TransactionItem[];
}

File: Models/Transaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Facit.Models
{
    public class Transaction
    {
        [Key]
        public int Id { get; set; }
        public Project Project { get; set; }
        public string Description { get; set; }
        public DateTime When { get; set; }
        public Person CreatedBy { get; set; }
        public Currency CurrencyLocal { get; set; }
        public IEnumerable<TransactionItem> TransactionItems { get; set; }
    }
}
